### DISCUSSION TITLE
Missing pixels in KAM after denosing

### DISCUSSION THREAD

**Discussion by voidnificance** [mtex-toolbox/mtex#2188] [Ask Anything]
Hi,

I was using both Kuwahara filter and the halfQuadraticFilter to denoising my dataset. Even though I tried to used the fill function during the smoothing process, it still loses some points(the phases of those points changes and the ebsd and ebsd('indexed') dataset becomes different). When I try to plot the KAM map, it shows several white pixels, is there a way to avoid those white pixels and fill the smoothed dataset to the original number of points? 

Here are my code:
```matlab
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'threshold',[0.5*degree, 2*degree]);
F = KuwaharaFilter;
F.numNeighbours = 3;
ebsd = smooth(ebsd('indexed'),F,'fill', grains);
ebsd = ebsd('indexed');

 plot(ebsd.gridify,ebsd.gridify.KAM('order',4) ./ degree);
```
![KAM3-new](https://github.com/user-attachments/assets/9b031b51-d621-43c0-bccb-2c2b07474289)

**Comment by Tijmenvermeij** [comment]
What probably happens here is that the gridify command reintroduces some empty areas in your non-gridded ebsd map. Maybe try the gridify command before applying the smoothening/filling?

**Comment by ralfHielscher** [comment]
You might want to try
```matlab
ebsd = ebsd('indexed')
[grains,ebsd.grainId] = calcGrains(ebsd,'threshold',[0.5*degree, 2*degree]);
ebsd(grains(grains.grainSize<=3))=[];
[grains,ebsd.grainId] = calcGrains(ebsd,'threshold',[0.5*degree, 2*degree]);
F = KuwaharaFilter;
F.numNeighbours = 3;
ebsd = smooth(ebsd,F,'fill');

plot(ebsd,ebsd.KAM('order',4) ./ degree);
```
Ralf.
